lib/rules/no-constant-condition.js

```diff
@@ -33,7 +33,10 @@ module.exports = {
 
     create(context) {
         const options = context.options[0] || {},
-            checkLoops = options.checkLoops !== false;
+            checkLoops = options.checkLoops !== false,
+            loopSetStack = [];
+
+        let setOfConstantLoops = new Set();
```

Comment:
Nitpick: Can you give this variable a more descriptive name? It's not immediately clear what loops are supposed to be contained in the set. (Something like `loopsInCurrentScope` seems like it would be a good name.)